xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/ExportAction.java

```diff
@@ -197,23 +220,69 @@ private String exportHTML(XWikiContext context) throws XWikiException, IOExcepti
                     for (String docName : docsNames) {
                         String pageReference = wikiName + XWikiDocument.DB_SPACE_SEP + docName;
                         if (context.getWiki().getRightService().hasAccessLevel(
-                            "view", context.getUser(), pageReference, context))
+                                "view", context.getUser(), pageReference, context))
                         {
                             pageList.add(resolver.resolve(pageReference));
                         }
                     }
                 }
             } catch (QueryException e) {
                 throw new XWikiException(XWikiException.MODULE_XWIKI_APP, XWikiException.ERROR_XWIKI_APP_EXPORT,
-                    "Failed to resolve pages to export", e);
+                        "Failed to resolve pages to export", e);
             } finally {
                 context.setWikiId(database);
             }
         }
-
         return pageList;
     }
 
+    /**
+     * Resolve the set of pages to export based on choices made in a paginated tree
+     * @param otherPages if true, the pagination was not completely resolved and the "other pages" checkbox
+     *                  was checked: the hidden pages should be taken into account
+     * @param checkedPages gives the name of the pages purposely selected (only used if otherPages is false)
+     * @param uncheckedPages gives the name of the pages purposely deselected (only used if otherPages is true)
+     * @param context the context of the request
+     * @return the collection of document reference to export
+     */
+    private Collection<DocumentReference> resolvePagesToExport(boolean otherPages, String[] checkedPages, String[] uncheckedPages, XWikiContext context)
+        throws XWikiException
+    {
+        if (otherPages) {
+            // if other pages is selected, we select all but the unchecked pages
+            XWikiDocument doc = context.getDoc();
+
+            // use the document tree to get all the pages
+            Tree nestedpages = Utils.getComponent(Tree.class, "nestedPages");
+            String nodeId = "document:" + doc.getDocumentReference().toString();
+            int childCount = nestedpages.getChildCount(nodeId);
+            List<String> children = nestedpages.getChildren(nodeId, 0, childCount);
```

Comment:
This gives you the direct child nodes. How do you include all the descendant child nodes? (I mean all the sub-tree).

Reply:
I suppose I don't right now: I did not check this scenario yet.
So 1. do you know if there's any way to get the descendant recursively from the tree API? 
   2. according to you, what is the minimum scenario to need an export with subtrees: defining a subwiki?

Reply:
OK I'll check what happens with nested pages then.

Reply:
Indeed there is a problem with nested pages, but I cannot process them as done in the javascript: in the JS we know which nodes are not loaded, so we can mark them with a joker to mark the subtree to be exported.
Here basically I need to process the nodes that are hidden: everyone of them might contain a subtree: so I would need to add the joker to everyone of them.